validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/ValidatorTypes.java

```diff
@@ -116,6 +119,48 @@
           .withField("data", listOf(REMOTE_KEY), Function.identity())
           .build();
 
+  public static DeserializableTypeDefinition<URL> URL_TYPE =
+      DeserializableTypeDefinition.string(URL.class)
+          .name("Signer")
+          .formatter(URL::toString)
+          .parser(
+              url -> {
+                try {
+                  return new URL(url);
+                } catch (MalformedURLException e) {
+                  throw new IllegalArgumentException(e);
+                }
+              })
+          .build();
+
+  public static DeserializableTypeDefinition<ExternalValidator> EXTERNAL_VALIDATOR_TYPE =
+      DeserializableTypeDefinition.object(ExternalValidator.class)
+          .name("ExternalValidator")
+          .initializer(ExternalValidator::new)
+          .withField(
+              "pubkey",
+              PUBKEY_TYPE,
+              ExternalValidator::getPublicKey,
+              ExternalValidator::setPublicKey)
+          .withOptionalField("url", URL_TYPE, ExternalValidator::getUrl, ExternalValidator::setUrl)
+          .withField(
+              "readonly",
+              BOOLEAN_TYPE,
+              ExternalValidator::isReadOnly,
+              ExternalValidator::setReadOnly)
+          .build();
```

Comment:
adding a test around this might be good too, just a positive test serializing and deserializing the single object.. if the deserialize requires a string for comparison like the list, it might be that the pubkey_type isn't quite right, as it should be a bytes comparison...

Reply:
I think i'm talking about making a roundtrip test - the string comparison is just because of the way the util works i think, so we currently dont really test a roundtrip of this object